### PR TITLE
GEODE-7864: Prevented non synchronized override of synchronized methods.

### PR DESCRIPTION
**Pull request by nabarunnag**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistTXPrecommitMessage.java [hunk 1]
```diff
@@ -315,7 +315,7 @@ public void waitForPrecommitCompletion() {
     }
 
     @Override
-    protected void processException(DistributionMessage msg, ReplyException ex) {
+    protected synchronized void processException(DistributionMessage msg, ReplyException ex) {
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/DistTXPrecommitMessage.java]: We may not need synchronized(this) in this method.

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistTXRollbackMessage.java [hunk 2]
```diff
@@ -311,7 +311,7 @@ public void waitForPrecommitCompletion() {
     }
 
     @Override
-    protected void processException(DistributionMessage msg, ReplyException ex) {
+    protected synchronized void processException(DistributionMessage msg, ReplyException ex) {
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/DistTXRollbackMessage.java]: We may not need synchronized(this) in this method.

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistTXPrecommitMessage.java (hunk 1 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/cache/DistTXPrecommitMessage.java:318]: The parents are synchronized and the override is not. This is not considered a good practice. I will look into this a bit more.

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistTXRollbackMessage.java (hunk 2 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/cache/DistTXRollbackMessage.java:314]: The parents are synchronized and the override is not. This is not considered a good practice. I will look into this a bit more.

**Pull request merged**